Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,41 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  [[ -z "$HOMEBREW_GIT" ]] && HOMEBREW_GIT="$(which_git 2>/dev/null)"
```

Comment:
But without it,  error here when git  is run in subshell will have strange result.